bundle/direct/bundle_apply.go

```diff
@@ -10,34 +10,45 @@ import (
 	"github.com/databricks/databricks-sdk-go"
 )
 
-func (b *DeploymentBundle) Apply(ctx context.Context, client *databricks.WorkspaceClient, configRoot *config.Root) {
+func (b *DeploymentBundle) Apply(ctx context.Context, client *databricks.WorkspaceClient, configRoot *config.Root, plan *deployplan.Plan) {
 	if b.Graph == nil {
 		panic("Planning is not done")
 	}
 
-	if len(b.DeploymentUnits) == 0 {
+	if len(plan.Plan) == 0 {
 		// Avoid creating state file if nothing to deploy
 		return
 	}
 
 	b.StateDB.AssertOpened()
 
 	b.Graph.Run(defaultParallelism, func(node deployplan.ResourceNode, failedDependency *deployplan.ResourceNode) bool {
-		d, exists := b.DeploymentUnits[node]
-		if !exists {
-			// Resource with actionType == noop are not added to DeploymentUnits.
-			// All references to it must have been resolved during planning.
+		key := "resources." + node.Group + "." + node.Key
```

Comment:
We seem to do this type of concatenation in multiple places, maybe worth abstracting it into `node.GetKey` func?